### PR TITLE
Fix create bottomup checkpoint

### PR DESCRIPTION
**Pull request by cryptoAtwill**
Fix the CI error in `create_bottom_up_checkpoint`: perform the gateway enabled check before doing contract queries.

Also fixing a minor issue in  submit checkpoint in relayer.

### CODE REVIEW

File: fendermint/vm/interpreter/src/fvm/checkpoint.rs [hunk 1]
```diff
@@ -322,23 +297,47 @@ fn should_create_checkpoint<DB>(
...
+    } else if height.value() % gateway.bottom_up_check_period(state)? == 0 {
+        tracing::debug!(
+            height = height.value(),
+            "bottom up checkpoint period reached height"
+        );
+    } else {
+        tracing::debug!(
+            height = height.value(),
+            "height has no message batch nor reached checkpoint period "
+        );
```
Comment (aakoshh) [inline fendermint/vm/interpreter/src/fvm/checkpoint.rs:331]: This will be a bit spammy on debug level at every block height.

File: fendermint/vm/interpreter/src/fvm/checkpoint.rs [hunk 2]
```diff
@@ -322,23 +297,47 @@ fn should_create_checkpoint<DB>(
...
+    let id = gateway.subnet_id(state)?;
+    let is_root = id.route.is_empty();
+
+    if is_root {
+        return Ok(None);
+    }
+
+    let batch = gateway.bottom_up_msg_batch(state, height.into())?;
+
+    if batch.block_height.as_u64() != 0 {
```
Comment (aakoshh) [inline fendermint/vm/interpreter/src/fvm/checkpoint.rs:317]: Have you considered some minimum batch size here?

File: fendermint/vm/interpreter/src/fvm/checkpoint.rs [hunk 3]
```diff
@@ -63,30 +63,11 @@ where
...
+    let (msgs, subnet_id) = if let Some(v) = should_create_checkpoint(gateway, state, height)? {
+        v
     } else {
-        tracing::debug!(
-            height = height.value(),
-            "bottom up msg batch found at height"
-        );
-    }
+        return Ok(None);
+    };
```
Comment (aakoshh) [inline fendermint/vm/interpreter/src/fvm/checkpoint.rs:70]: You can use the new [let-else](https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html) syntax here:

```rust
let Some((msgs, subnet_id)) = should_create_checkpoint(...) else {
  return Ok(None);
}
```

File: ipc/provider/src/checkpoint.rs [hunk 4]
```diff
@@ -199,18 +199,22 @@ impl<T: BottomUpCheckpointRelayer + Send + Sync + 'static> BottomUpCheckpointMan
                     .await?;
                 log::debug!("bottom up bundle: {bundle:?}");
 
-                todo!("implement submit checkpoint")
```
Comment (aakoshh) [inline ipc/provider/src/checkpoint.rs:202]: This change look unrelated from the fix. Just to be clear this is called by the relayer, right?

**Review by aakoshh — APPROVED** [review]
Looks correct, just minor comments.

File: ipc/provider/src/checkpoint.rs (hunk 4 above)
Comment (cryptoAtwill) [reply ipc/provider/src/checkpoint.rs:202]: yeah, it's a minor thing, should have enabled, but too small for another PR.

File: fendermint/vm/interpreter/src/fvm/checkpoint.rs (hunk 1 above)
Comment (cryptoAtwill) [reply fendermint/vm/interpreter/src/fvm/checkpoint.rs:331]: yeah, probably the batch full log message can stay, this one can go

File: fendermint/vm/interpreter/src/fvm/checkpoint.rs (hunk 2 above)
Comment (cryptoAtwill) [reply fendermint/vm/interpreter/src/fvm/checkpoint.rs:317]: From the contract, there is no minimum batch size, it's either non-exists or some number of messages. If it's zero, that means the batch does not exists.

Comment (aakoshh) [reply fendermint/vm/interpreter/src/fvm/checkpoint.rs:315]: I know, I meant: should it create a new checkpoint even if there is just 1 message in the batch, or should it wait until there is some, but not too many; e.g. make a checkpoint whenever there are at least 10 outgoing messages _or_ 1000 blocks have been created.
